### PR TITLE
Fix tcpdump detection

### PR DESCRIPTION
**Pull request by gpotter2**
fixes https://github.com/secdev/scapy/issues/2223

### CODE REVIEW

**Comment by guedou** [comment]
Please don’t merge until tested on OpenBSD.

File: scapy/arch/common.py [hunk 1]
```diff
@@ -32,18 +31,19 @@ def _check_tcpdump():
...
+        proc = subprocess.Popen(
+            [conf.prog.tcpdump, "--version"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT
+        )
+        output = proc.communicate()[0]
+    except OSError:
+        return False
+
+    # tcpdump acts strangely on some OSes and returns 1
```
Comment (p-l-) [inline scapy/arch/common.py]:
```suggestion
```

File: scapy/arch/common.py [hunk 2]
```diff
@@ -32,18 +31,19 @@ def _check_tcpdump():
...
+    # therefore we also checks the output
```
Comment (p-l-) [inline scapy/arch/common.py]:
```suggestion
    # On some systems, --version does not exist on tcpdump
```

File: scapy/arch/common.py (hunk 1 above)
Comment (p-l-) [inline scapy/arch/common.py]:
```suggestion
```

File: scapy/arch/common.py [hunk 3]
```diff
@@ -32,18 +31,19 @@ def _check_tcpdump():
...
+    return b"tcpdump" in output or proc.returncode == 0
```
Comment (p-l-) [inline scapy/arch/common.py]: I would inverse the condition and be more precise on the second case:
```suggestion
    return proc.returncode == 0 or output.startswith(b'Usage: tcpdump ')
```

**Pull request merged**